test/scale/scale_tester.go

```diff
@@ -178,94 +186,117 @@ scaleInTest:
 }
 
 func (t *PodSchedulingOnFailoverScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
-	statefulSets := []*resources.TestStatefulset{}
+	totalNumberOfPods := t.Replicas
+
+	nodes := nodeutil.ListAgentNodeNames(client, t.Pod.IsWindows)
+	skipper.SkipUnlessAtLeast(len(nodes), 2, "Need at least 2 agent nodes to verify the test case.")
+	midNode := len(nodes) / 2
+
+	testPod := resources.TestPod{
+		Client: client,
+	}
 
-	ginkgo.By("Initiating statefulset setup")
+	// uncordon the first half nodes
+	for _, node := range nodes[:midNode] {
+		node := node
+		framework.Logf("Uncordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, false)
```

Comment:
Instead of having `SetNodeUnschedulable` a menthod of `testPod` let's make it a function in `test/resources/utils.go` and pass the `client` as a parameter.

Reply:
Yeah, it makes more sense to move the function to `node_utils.go`. I'll get rid of the existing one.